docs/development/how_it_works.md

```diff
@@ -0,0 +1,132 @@
+# How it works
+
+## 1) The CLI
+
+Everything in `kops` is currently driven by a command line interface. We use [cobra](https://github.com/spf13/cobra) to define all of our command line UX.
+
+All of the CLI code for kops can be found in `/cmd/kops` [link](https://github.com/kubernetes/kops/tree/master/cmd/kops)
+
+For instance, if you are interested in finding the entry point to `kops create cluster` you would look in `/cmd/kops/create_cluster.go`. There you would find a function called `RunCreateCluster()`. That is the entry point of the command.
+
+## 2) Storage
+
+We have an abstracted way of interacting with our storage, called the `clientset`. This is an ambiguous way of interacting with remote storage for `kops`. This is commonly referred to as the `kops` **STATE STORE**.
+
+We can access it through the `util.Factory` struct as in :
+
+```Go
+func RunMyCommand(f *util.Factory, out io.Writer, c *MyCommandOptions) error {
+        clientset, _ := f.Clientset()
+        cluster, _ := clientset.Clusters().Get(clusterName)
+        fmt.Println(cluster)
+}
+```
+
+The available Clientset functions are :
+
+```
+Create()
+Update()
+Get()
+List()
+```
+## 3) The API
+
+#### a) Cluster Spec
+
+The `kops` API is a definition of struct members in Go code found [here](https://github.com/kubernetes/kops/tree/master/pkg/apis/kops). The `kops` API does **NOT** match the command line interface (by design). We use the native Kubernetes API machinery to manage versioning the `kops` API.
+
+The base level struct of the API is `api.Cluster{}` which is defined [here](https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/cluster.go#L40). The top level struct contains meta information about the API we are currently using, and the main definition of the cluster itself can be found in `cluster.Spec`
```

Comment:
s/the API we are currently using,/the object such as the kind and version,/g

s/main definition of the cluster/main data for the cluster/g